lib/matplotlib/tests/test_legend.py

```diff
@@ -1219,3 +1219,71 @@ def test_ncol_ncols(fig_test, fig_ref):
     ncols = 3
     fig_test.legend(strings, ncol=ncols)
     fig_ref.legend(strings, ncols=ncols)
+
+
+def test_loc_invalid_tuple_exception():
+    # check that exception is raised if the loc arg
+    # of legend is not a 2-tuple of numbers
+    fig, ax = plt.subplots()
+    with pytest.raises(ValueError) as errorinfo:
```

Comment:
You can use the `match` keyword in `pytest.raises` to specify the expected error message.
https://docs.pytest.org/en/7.2.x/reference/reference.html#pytest.raises